python-checks/src/main/java/org/sonar/python/checks/cdk/S3BucketServerEncryptionCheck.java

```diff
@@ -0,0 +1,119 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.cdk;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.PythonSubscriptionCheck;
+import org.sonar.plugins.python.api.SubscriptionContext;
+import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.Argument;
+import org.sonar.plugins.python.api.tree.CallExpression;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.Name;
+import org.sonar.plugins.python.api.tree.QualifiedExpression;
+import org.sonar.plugins.python.api.tree.RegularArgument;
+import org.sonar.plugins.python.api.tree.Tree;
+import org.sonar.python.checks.Expressions;
+
+@Rule(key = "S6245")
+public class S3BucketServerEncryptionCheck extends PythonSubscriptionCheck {
+
+  public static final String MESSAGE = "Objects in the bucket are not encrypted. Make sure it is safe here.";
+  public static final String MESSAGE_OMITTING = "Omitting 'encryption' disables server-side encryption. Make sure it is safe here.";
+  public static final String MESSAGE_SECONDARY = "Propagated setting.";
+
+  @Override
+  public void initialize(Context context) {
+    context.registerSyntaxNodeConsumer(Tree.Kind.CALL_EXPR, this::visitNode);
+  }
+
+  public void visitNode(SubscriptionContext ctx) {
+    CallExpression node = (CallExpression) ctx.syntaxNode();
+    Optional.ofNullable(node.calleeSymbol())
+      .filter(nodeSymbol -> "aws_cdk.aws_s3.Bucket".equals(nodeSymbol.fullyQualifiedName()))
+      .ifPresent(nodeSymbol -> {
+        List<RegularArgument> args = getEncryptionArguments(node.arguments());
+        Optional<RegularArgument> optEncryptionType = args.stream().filter(a -> "encryption".equals(getArgumentName(a))).findFirst();
+        Optional<RegularArgument> optEncryptionKey = args.stream().filter(a -> "encryption_key".equals(getArgumentName(a))).findFirst();
+        if (optEncryptionType.isPresent()) {
+          optEncryptionType.map(regArg -> propagatedSensitiveExpressions(regArg.expression()))
+            .filter(trees -> !trees.isEmpty())
+            .ifPresent(trees -> {
+              PreciseIssue issue = ctx.addIssue(trees.get(0).parent(), MESSAGE);
+              trees.stream().skip(1).forEach(expression -> issue.secondary(expression.parent(), MESSAGE_SECONDARY));
+            });
+        } else {
+          if (!optEncryptionKey.isPresent()) {
+            ctx.addIssue(node.callee(), MESSAGE_OMITTING);
+          }
+        }
+      });
+  }
+
+  private static List<RegularArgument> getEncryptionArguments(List<Argument> args) {
```

Comment:
```suggestion
  private static RegularArgument getEncryptionArguments(List<Argument> args, String argName) {
```
This way you simplify also the logic of `visitNode`